packages/react-router-dev/vite/plugin.ts

```diff
@@ -1462,7 +1461,9 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = (_config) => {
         let [filepath] = id.split("?");
 
         let ast = parse(code, { sourceType: "module" });
-        removeExports(ast, SERVER_ONLY_ROUTE_EXPORTS);
+        if (!options?.ssr) {
+          removeExports(ast, SERVER_ONLY_ROUTE_EXPORTS);
+        }
```

Comment:
This is the important part: the `WithProps.transform` (below) always needs to be applied so that SSR render also has data from props HOCs, which means we need to conditionally remove exports for `ssr: false` more precisely.